bundler/spec/quality_spec.rb

```diff
@@ -253,9 +253,62 @@ def check_for_specific_pronouns(filename)
     expect(lib_code).to eq(spec_code)
   end
 
+  it "documents all cli command options in their associated man pages" do
+    commands = normalize_commands_and_options(Bundler::CLI)
+    cli_and_man_pages_in_sync!(commands)
+
+    Bundler::CLI.subcommand_classes.each do |_, klass|
+      subcommands = normalize_commands_and_options(klass)
+
+      cli_and_man_pages_in_sync!(subcommands)
+    end
+  end
+
   private
 
   def each_line(filename, &block)
     File.readlines(filename, encoding: "UTF-8").each_with_index(&block)
   end
+
+  def normalize_commands_and_options(command_class)
+    commands = {}
+
+    command_class.commands.each do |command_name, command|
+      next if command.is_a?(Bundler::Thor::HiddenCommand)
+
+      key = command.ancestor_name || command_name
+      commands[key] ||= []
+      # Verify that all subcommands are documented in the main command's man page.
+      commands[key] << command_name unless command_class == Bundler::CLI
+
+      command.options.each do |name, option|
+        commands[key] << name unless option.banner.to_s.include?("deprecated")
+      end
+    end
+
+    commands
+  end
+
+  def cli_and_man_pages_in_sync!(commands)
+    undocumented_options = ["help", "keep_file_descriptors", "all"]
```

Comment:
I'm happy to add documentation for the `bundle exec --keep_file_descriptors` and `bundle cache --all`, I just wanted to check first if this implementation would get accepted.